packages/ai-jsx/src/lib/openai.tsx

```diff
@@ -45,17 +45,45 @@ export type ValidChatModel =
   | 'gpt-3.5-turbo-16k'
   | 'gpt-3.5-turbo-16k-0613';
 
+/**
+ * An OpenAI client that talks to the Azure OpenAI service.
+ * - Adds an API version in the query string.
+ * - Uses Api-Key instead of Authorization for the auth header.
+ * - Includes the deployment name in the path; note that deployment names cannot contain dots.
+ */
+class AzureOpenAIClient extends OpenAIClient {
+  protected override defaultQuery() {
+    return { 'api-version': '2023-07-01-preview' };
+  }
+  protected override authHeaders() {
+    return {
+      'Api-Key': this.apiKey,
+    };
+  }
+  override buildRequest(options: FinalRequestOptions) {
+    const result = super.buildRequest(options);
+    if (options.body && 'model' in options.body) {
+      const model = (options.body.model as string).replace('.', '');
+      result.url = result.url.replace('/chat/', `/deployments/${model}/chat/`);
```

Comment:
For extra safety, you could use the [url](https://nodejs.org/api/url.html) module to parse out the `pathname` and only run the `replace` on that.

But I guess the `/` characters can only appear in the pathname, so this probably will not have an impact in practice.

Reply:
yeah good call, I found a cleaner approach that isn't isolated to /chat/, so I went with that.